cmd/alias.go

```diff
@@ -46,10 +46,9 @@ func (c *AliasCommand) Run(args []string) int {
 
 	args = c.flags.Args()
 
-	switch len(args) {
-	case 0:
-	default:
-		c.UI.Error(fmt.Sprintf("Error: too many arguments (expected 0, got %d)\n", len(args)))
+	argCountErr := validateArgumentCount(args, 0, 0)
+	if argCountErr != nil {
```

Comment:
What if we made this a struct with methods too? It could look something like:

```go
type CommandArgumentValidator struct {
  args []string{}
  required Int
  optional Int
  // or maybe initialize with UI too
}

func (c *CommandArgumentValidator) validate(ui cli.Ui) int {
  # etc
  ui.Error(argCountErr.Error())
  ui.Output(c.Help())
  return 1
}

// in a command

validator := CommandArgumentValidator{args: args, required: 0, optional: 0}
validator.validate(c.UI)
```

One reason I like this is for the named parameters

Reply:
Updated.

Usage:

```golang
commandArgumentValidator := &CommandArgumentValidator{required: 1, optional: 1}
commandArgumentErr := commandArgumentValidator.validate(args)
if commandArgumentErr != nil {
	c.UI.Error(commandArgumentErr.Error())
	c.UI.Output(c.Help())
	return 1
}
```

---

To keep `CommandArgumentValidator` simple., I am not passing `c.UI` & `c.Help()` to `CommandArgumentValidator`.

Polluting `CommandArgumentValidator` with `c.UI` & `c.Help()` to make 2 less lines of code doesn't worth and looks like a violation of single responsibility principle to me. :
```goland
commandArgumentValidator := &CommandArgumentValidator{required: 1, optional: 1, UI: c.UI, Help: c.Help()}
commandArgumentErr := commandArgumentValidator.validate(args)
if commandArgumentErr != nil {
	return 1
}
```

What do you think?

Reply:
I think I agree 👍